core/src/main/java/org/apache/calcite/avatica/remote/looker/LookerRemoteMeta.java

```diff
@@ -0,0 +1,375 @@
+   */
+  private void trustAllHosts(HttpsURLConnection connection) {
+    // Create a trust manager that does not validate certificate chains
+    TrustManager[] trustAllCerts = new TrustManager[]{
+        new X509TrustManager() {
+          public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+            return null;
+          }
+          public void checkClientTrusted(java.security.cert.X509Certificate[] certs,
+              String authType) {
+          }
+          public void checkServerTrusted(java.security.cert.X509Certificate[] certs,
+              String authType) {
+          }
+        }
+    };
+    // Create all-trusting host name verifier
+    HostnameVerifier trustAllHostNames = (hostname, session) -> true;
+    try {
+      SSLContext sc = SSLContext.getInstance("SSL");
+      sc.init(null, trustAllCerts, new java.security.SecureRandom());
+      connection.setSSLSocketFactory(sc.getSocketFactory());
+      connection.setHostnameVerifier(trustAllHostNames);
+    } catch (NoSuchAlgorithmException | KeyManagementException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * A regrettable necessity. The Kotlin SDK relies on an outdated Ktor HTTP client based on Kotlin
+   * Coroutines which are a tad difficult to work with in Java. Here we make a HTTP client to handle
+   * the response stream ourselves which is "good enough" for now. This method should be revisited
+   * when the Kotlin SDK has built-in streams.
+   *
+   * TODO https://github.com/looker-open-source/sdk-codegen/issues/1341:
+   *  Add streaming support to the Kotlin SDK.
+   */
+  private void streamResponse(String url, OutputStream outputStream) throws IOException {
+    // use some SDK client helpers to tighten up this call
+    AuthSession authSession = getSdk().getAuthSession();
+    Transport sdkTransport = authSession.getTransport();
+    // makes a proper URL from the API endpoint path as the SDK would.
+    String endpoint = sdkTransport.makeUrl(url, Collections.emptyMap(), null);
+    URL httpsUrl = new URL(endpoint);
+    HttpsURLConnection connection = (HttpsURLConnection) httpsUrl.openConnection();
+    // WARNING: You should only set `verifySSL=false` for local/dev instances!!
+    if (!sdkTransport.getOptions().getVerifySSL()) {
+      trustAllHosts(connection);
+    }
+    connection.setRequestMethod("POST");
+    connection.setDoOutput(true);
+    // Set the auth header as the SDK would
+    connection.setRequestProperty("Authorization",
+        "token " + authSession.getAuthToken().getAccessToken());
+    connection.setRequestProperty("Accept", "application/json");
+    int responseCode = connection.getResponseCode();
+    if (responseCode == 200) {
+      // grab the input stream and write to the output for the main thread to consume.
+      InputStream inputStream = connection.getInputStream();
+      int bytesRead;
+      byte[] buffer = new byte[DEFAULT_STREAM_BUFFER_SIZE];
```

Comment:
This buffer size doesn't control the actual buffer (byte window of I/O concurrency) between the HTTP connection and the client iterating over `LookerIterable`. You're not guaranteed to get a full buffer-worth from HTTP / TCP transport, and the pipe only has a tiny buffer of 1024 by default. If the client is slow at iterating, it may have to wait for network transport every time. That may be an acceptable design, I don't know what the design parameters are around slow clients.